drivers/sensor/pixart/pat9136/pat9136_stream.c

```diff
@@ -0,0 +1,431 @@
+				      data->stream.settings.enabled.motion;
+	edata->header.channels = 0;
+
+	if ((data->stream.settings.enabled.drdy &&
+	      data->stream.settings.opt.drdy == SENSOR_STREAM_DATA_INCLUDE) ||
+	    (data->stream.settings.enabled.motion &&
+	      data->stream.settings.opt.motion == SENSOR_STREAM_DATA_INCLUDE)) {
+		edata->header.channels |= pat9136_encode_channel(SENSOR_CHAN_POS_DXYZ);
+		edata->header.channels |= pat9136_encode_channel(SENSOR_CHAN_POS_DXYZ_MM);
+	}
+
+	if (data->stream.settings.enabled.drdy) {
+		start_drdy_backup_timer(dev);
+	}
+
+	/** Attempt chip recovery if erratic behavior is detected  */
+	if (!REG_OBSERVATION_READ_IS_VALID(edata->observation)) {
+
+		LOG_WRN("CHIP OK register indicates issues. Attempting chip recovery: 0x%02X",
+			edata->observation);
+
+		rtio_iodev_sqe_err(iodev_sqe, -EAGAIN);
+	} else {
+		rtio_iodev_sqe_ok(iodev_sqe, 0);
+	}
+}
+
+static void pat9136_stream_get_data(const struct device *dev)
+{
+	struct pat9136_data *data = dev->data;
+	uint64_t cycles;
+	int err;
+
+	CHECKIF(!data->stream.iodev_sqe) {
+		LOG_WRN("No RTIO submission with an INT GPIO event");
+		return;
+	}
+
+	start_drdy_cooldown_timer(dev);
+
+	struct pat9136_encoded_data *buf;
+	uint32_t buf_len;
+	uint32_t buf_len_required = sizeof(struct pat9136_encoded_data);
+
+	err = rtio_sqe_rx_buf(data->stream.iodev_sqe,
+			      buf_len_required,
+			      buf_len_required,
+			      (uint8_t **)&buf,
+			      &buf_len);
+	__ASSERT(err == 0, "Failed to acquire buffer (len: %d) for encoded data: %d. "
+			   "Please revisit RTIO queue sizing and look for "
+			   "bottlenecks during sensor data processing",
+			   buf_len_required, err);
+
+	/** Still throw an error even if asserts are off */
+	if (err) {
+		struct rtio_iodev_sqe *iodev_sqe = data->stream.iodev_sqe;
+
+		LOG_ERR("Failed to acquire buffer for encoded data: %d", err);
+
+		data->stream.iodev_sqe = NULL;
+		rtio_iodev_sqe_err(iodev_sqe, err);
+		return;
+	}
+
+	struct rtio_sqe *write_res_x_sqe = rtio_sqe_acquire(data->rtio.ctx);
+	struct rtio_sqe *read_res_x_sqe = rtio_sqe_acquire(data->rtio.ctx);
+	struct rtio_sqe *write_res_y_sqe = rtio_sqe_acquire(data->rtio.ctx);
+	struct rtio_sqe *read_res_y_sqe = rtio_sqe_acquire(data->rtio.ctx);
+	struct rtio_sqe *write_sqe = rtio_sqe_acquire(data->rtio.ctx);
+	struct rtio_sqe *read_sqe = rtio_sqe_acquire(data->rtio.ctx);
+	struct rtio_sqe *cb_sqe = rtio_sqe_acquire(data->rtio.ctx);
+	uint8_t val;
+
+	err = sensor_clock_get_cycles(&cycles);
+	CHECKIF(err) {
+		struct rtio_iodev_sqe *iodev_sqe = data->stream.iodev_sqe;
+
+		LOG_ERR("Failed to get timestamp: %d", err);
+
+		data->stream.iodev_sqe = NULL;
+		rtio_iodev_sqe_err(iodev_sqe, err);
+		return;
+	}
+	buf->header.timestamp = sensor_clock_cycles_to_ns(cycles);
+
+	CHECKIF(!write_res_x_sqe || !read_res_x_sqe ||
+		!write_res_y_sqe || !read_res_y_sqe ||
+		!write_sqe || !read_sqe || !cb_sqe) {
+		LOG_ERR("Failed to acquire RTIO SQE's");
+
```

Comment:
rtio_sqe_drop_all(data->rtio.ctx); here if you want to try and recover later, otherwise any acquired sqe's above will forever be in limbo.

Reply:
Good catch. I've added it now